lua/orgmode/objects/date.lua

```diff
@@ -490,12 +525,16 @@ end
 
 ---@return number
 function Date:get_isoweekday()
+  ---@type table
+  ---@diagnostic disable-next-line: assign-type-mismatch
   local date = os.date('*t', self.timestamp)
   return utils.convert_to_isoweekday(date.wday)
 end
 
 ---@return number
 function Date:get_weekday()
+  ---@type table
+  ---@diagnostic disable-next-line: assign-type-mismatch
```

Comment:
We can add `wday` to `OsDate` type and use `os_date` here.

Reply:
Done.